docs: Add Missing SSR integrity marker error doc page

JeanMeche opened the pull request:
This commit adds an error doc page for error 507 : Missing SSR content integrity marker.

AndrewKushnir commented on aio/content/errors/NG0507.md:
```diff
@@ -0,0 +1,16 @@
+@shortDescription Angular hydration logic detected that HTML content of this page was modified
+
+@description
+Angular hydration relies on HTML produced by the server, including whitespaces and comment nodes.
```
```suggestion
Angular throws this error when it detects that the content generated by server-side rendering (SSR) was altered after the rendering. The process of hydration relies on the content to be untouched after SSR, which also includes whitespaces and comment nodes. Those whitespaces and comment nodes must be retained in the HTML generated by the SSR process. Learn more in the [Hydration guide](guide/hydration#constraints).
```

AndrewKushnir commented on aio/content/errors/NG0507.md:
```diff
@@ -0,0 +1,16 @@
+@description
+Angular hydration relies on HTML produced by the server, including whitespaces and comment nodes.
+
+This errors occurs when the Angular hydration logic detects that HTML content generated service side was modified after SSR is completed. For example, when the whitespaces and comment nodes are stripped off from the HTML. 
```
(this is covered by the paragraph above)
```suggestion
```

AndrewKushnir commented on aio/content/errors/NG0507.md:
```diff
@@ -0,0 +1,16 @@
+
+@debugging
+
+Typically this happens either by CDN or during the build process as an optimization to remove comment nodes.
+
+Make sure that there are no optimizations that remove comment nodes from HTML enabled on your CDN.
+
```
```suggestion
Typically this happens in the following cases:

* Some CDN providers have a built-in feature to remove whitespaces and comment nodes from HTML as an optimization. Please verify if there is such an option in CDN configuration and turn it off.
* If you use custom post-processing of HTML generated by SSR (as a build step), make sure that this process doesn't remove whitespaces and comment nodes.

```

AndrewKushnir reviewed (commented):
@JeanMeche thanks for creating this PR! 👍 

The change looks good, I've just added a few suggestions.

JeanMeche committed:
docs: Add Missing SSR integrity marker error doc page

This commit adds an error doc page for error 507 : Missing SSR content integrity marker.

AndrewKushnir reviewed (approved):
👍

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

AndrewKushnir commented:
**Caretaker note**: this is a low risk change, no extra reviews are required.

jessicajaniuk reviewed (approved):
reviewed-for: public-api

pkozlowski-opensource commented:
This PR was merged into the repository by commit 3b1c1b91a2c1d8104ed5deb650e680ae237c97d8.
